Description: Enable build on s390x
 Note that this is only part of the fix.  We also patch configure.in
 prior to running autoreconf and generating the .orig.tar.gz tarball.
 See debian/rules and orig-source.patch.
Author: Philipp Kern <pkern@debian.org> 
Bug: 635461 
Forwarded: http://gmplib.org/list-archives/gmp-bugs/2011-August/002346.html


--- gmp-5.0.1+dfsg.orig/mpn/s390x/gmp-mparam.h
+++ gmp-5.0.1+dfsg/mpn/s390x/gmp-mparam.h
@@ -0,0 +1,28 @@
+/*  IBM S/390 64 bit gmp-mparam.h -- Compiler/machine parameter header file.
+
+Copyright (C) 2009 Free Software Foundation, Inc.
+
+This file is part of the GNU MP Library.
+
+The GNU MP Library is free software; you can redistribute it and/or modify
+it under the terms of the GNU Library General Public License as published by
+the Free Software Foundation; either version 2 of the License, or (at your
+option) any later version.
+
+The GNU MP Library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Library General Public
+License for more details.
+
+You should have received a copy of the GNU Library General Public License
+along with the GNU MP Library; see the file COPYING.LIB.  If not, write to
+the Free Software Foundation, Inc., 59 Temple Place - Suite 330, Boston,
+MA 02111-1307, USA. */
+
+#define BITS_PER_MP_LIMB 64
+#define BYTES_PER_MP_LIMB 8
+
+#define BITS_PER_LONGINT 64
+#define BITS_PER_INT 32
+#define BITS_PER_SHORTINT 16
+#define BITS_PER_CHAR 8
only in patch2:
unchanged:
--- gmp-5.0.1+dfsg.orig/mpn/s390/gmp-mparam.h
+++ gmp-5.0.1+dfsg/mpn/s390/gmp-mparam.h
@@ -17,9 +17,13 @@
 You should have received a copy of the GNU Lesser General Public License
 along with the GNU MP Library.  If not, see http://www.gnu.org/licenses/.  */
 
+#define BITS_PER_MP_LIMB 32
+#define BYTES_PER_MP_LIMB 4
 
-/* GMP_LIMB_BITS etc generated by configure */
-
+#define BITS_PER_LONGINT 32
+#define BITS_PER_INT 32
+#define BITS_PER_SHORTINT 16
+#define BITS_PER_CHAR 8
 
 /* Generated by tuneup.c, 2001-12-03, gcc 2.95 */
 
